hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/CleanFunction.java

```diff
@@ -60,11 +60,13 @@ public CleanFunction(Configuration conf) {
   @Override
   public void open(Configuration parameters) throws Exception {
     super.open(parameters);
-    this.writeClient = FlinkWriteClients.createWriteClient(conf, getRuntimeContext());
-    this.executor = NonThrownExecutor.builder(LOG).waitForTasksFinish(true).build();
-    String instantTime = HoodieActiveTimeline.createNewInstantTime();
-    LOG.info(String.format("exec clean with instant time %s...", instantTime));
-    executor.execute(() -> writeClient.clean(instantTime), "wait for cleaning finish");
+    if (conf.getBoolean(FlinkOptions.CLEAN_ASYNC_ENABLED)) {
+      this.writeClient = FlinkWriteClients.createWriteClient(conf, getRuntimeContext());
+      this.executor = NonThrownExecutor.builder(LOG).waitForTasksFinish(true).build();
+      String instantTime = HoodieActiveTimeline.createNewInstantTime();
+      LOG.info(String.format("exec clean with instant time %s...", instantTime));
+      executor.execute(() -> writeClient.clean(instantTime), "wait for cleaning finish");
+    }
```

Comment:
nit: we can add a line of log here to indicate to the user that async clean is disabled although the clean function is opened. 

LOG.info("Async clean is disabled");